modules/imgcodecs/src/grfmt_webp.cpp

```diff
@@ -126,8 +86,38 @@ bool WebPDecoder::readHeader()
     WebPBitstreamFeatures features;
     if (VP8_STATUS_OK == WebPGetFeatures(header, sizeof(header), &features))
     {
-        CV_CheckEQ(features.has_animation, 0, "WebP backend does not support animated webp images");
+        m_has_animation = features.has_animation > 0;
+        if (m_has_animation)
+        {
+            fs.seekg(0, std::ios::beg); CV_Assert(fs && "File stream error");
+            data.create(1, validateToInt(fs_size), CV_8UC1);
+            fs.read((char*)data.ptr(), fs_size);
+            CV_Assert(fs && "Can't read file data");
+            fs.close();
+
+            CV_Assert(data.type() == CV_8UC1); CV_Assert(data.rows == 1);
+
+            WebPData webp_data;
+            webp_data.bytes = (const uint8_t*)data.ptr();
+            webp_data.size = data.total();
 
+            WebPAnimDecoderOptions dec_options;
+            WebPAnimDecoderOptionsInit(&dec_options);
+            dec_options.color_mode = MODE_BGRA;
+            anim_decoder = WebPAnimDecoderNew(&webp_data, &dec_options);
```

Comment:
You are missing a WebPAnimDecoderDelete. Either you create a destructor for WebPDecoder, either you wrap the pointer in a unique_ptr. E.g., create in the header, a private struct:
```cpp
struct UniquePtrDeleter
{
    void operator()(WebPAnimDecoder * decoder) const { WebPAnimDecoderDelete(decoder); }
    void operator()(WebpData * dara) const { WebPDataClear(decoder); }
};

// Use these unique_ptr to ensure the structs are automatically destroyed.
std::unique_ptr<avifEncoder, UniquePtrDeleter> anim_decoder;
// or create a type:
using DataPtr = std::unique_ptr<WebPData, UniquePtrDeleter>;
```
I'd go with this, so that you do not have to remember to call a destructor.

Reply:
@vrabaud could you check the last commit. frankly i tried your suggestion using std::unique_ptr but no success.. can last commit resolve this conversation.

Reply:
When calling the WebP API you need to get a pointer so if you have a `std::unique_ptr<WebPAnimDecoder, UniquePtrDeleter> anim_decoder;`, just call `anim_decoder.get()`

Reply:
@vrabaud could you check https://github.com/opencv/opencv/pull/25608/commits/a105d94aff541d2b4593150dc834bab279ad6cd6

Reply:
should i mark this as resolved?

Reply:
Just do the same thing for data and encoder (or maybe simply do what you did for the encoder, by using a full  unique_ptr definition if you only use those once in the code.